src/hermezjs-typings.d.ts

```diff
@@ -0,0 +1,731 @@
+   *
+   * Some types and props are currently commented because are not used.
+   * As we proceed with the migration of the application to typescript,
+   * more properties and types will likely be required. We can uncomment
+   * them as required and finally remove all those not required.
+   */
+
+  export type ISOStringDate = string;
+
+  export interface HermezStatus {
+    isUnderMaintenance: boolean;
+  }
+
+  export type HermezNetworkStatus = "online" | "offline";
+
+  export interface Wallet {
+    publicKeyBase64: string;
+    publicKeyCompressedHex: string;
+    hermezEthereumAddress: string;
+    // privateKey: Buffer;
+    // publicKey: string[];
+    // publicKeyHex: string[];
+    // publicKeyCompressed: string;
+  }
+
+  export interface Signer {
+    address: string;
+    type: string;
+  }
+
+  // ToDo: Consider explicitly setting the supported FIAT
+  export type FiatExchangeRates = Record<string, number>;
+
+  export interface HermezApiResourceItem {
+    itemId: number;
+  }
+
+  export type Token = HermezApiResourceItem & {
+    decimals: number;
+    ethereumAddress: string;
+    ethereumBlockNum: number;
+    fiatUpdate: ISOStringDate;
+    id: number;
+    name: string;
+    symbol: string;
+    USD: number;
+  };
+
+  export interface MerkleProof {
+    root: string;
+    siblings: BigNumber[];
+    // oldKey: string;
+    // oldValue: string;
+    // isOld0: boolean;
+    // key: string;
+    // value: string;
+    // fnc: number;
+  }
+
+  export type Exit = HermezApiResourceItem & {
+    batchNum: number;
+    accountIndex: string;
+    // bjj: string;
+    // hezEthereumAddress: string;
+    // merkleProof: MerkleProof;
+    // balance: string;
+    instantWithdraw: number;
+    // delayedWithdrawRequest: unknown | null;
+    delayedWithdraw: unknown | null;
+    // token: Token;
+  };
+
+  export interface Withdraw {
+    accountIndex: string;
+    // amount: string;
+    batchNum: number;
+    hash: string;
+    // hermezEthereumAddress: string;
+    id: string;
+    timestamp: ISOStringDate;
+    // token: Token;
+  }
+
+  export type DelayedWithdraw = Withdraw & {
+    // merkleProof: MerkleProof;
+    instant: boolean;
+  };
+
+  export interface Deposit {
+    // account: Account;
+    hash: string;
+    // fromHezEthereumAddress: string;
+    // toHezEthereumAddress: string;
+    token: Token;
+    amount: string;
+    timestamp: ISOStringDate;
+    // ToDo: This states are yet to be validated
+    // state: "fged" | "fing" | "pend" | "invl";
+    // ToDo: Are there more deposit types??
+    type: "Deposit" | "CreateAccountDeposit";
+  }
+
+  // ToDo: What is the shape of a reward?
+  export type Reward = unknown;
+
+  export type Account = HermezApiResourceItem & {
+    accountIndex: string;
+    balance: string;
+    bjj: string;
+    fiatBalance: number;
+    // hezEthereumAddress: string;
+    // nonce: number;
+    token: Token;
+  };
+
+  export type L2Transaction = HermezApiResourceItem & {
```

Comment:
Now that we don't have the `L1Transaction` this could be called just `Transaction`.

Reply:
> Now that we don't have the L1Transaction

I still don't see clear. Let's meet and discuss it.

Reply:
Done [here](#discussion_r705201827).